File: app-java/src/main/java/net/mEmoZz/yts/java/ui/detail/adapters/recycler/CastAdapter.java
package net.mEmoZz.yts.java.ui.detail.adapters.recycler;

import android.content.Context;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;
import butterknife.BindView;
import butterknife.ButterKnife;
import de.hdodenhof.circleimageview.CircleImageView;
import java.util.List;
import net.mEmoZz.yts.java.R;
import net.mEmoZz.yts.java.data.models.MovieDetail;
import net.mEmoZz.yts.java.utilities.GlideUtil;
import net.mEmoZz.yts.java.utilities.Utils;

/**
 * Authored by Mohamed Fathy on 08 Mar, 2018.
 * Contact: muhamed.gendy@gmail.com
 */

public class CastAdapter extends RecyclerView.Adapter<CastAdapter.CastHolder> {

  private Context context;
  private List<MovieDetail.Cast> casts;

  public CastAdapter(Context context, List<MovieDetail.Cast> casts) {
    this.context = context;
    this.casts = casts;
  }

  @Override public CastHolder onCreateViewHolder(ViewGroup parent, int viewType) {
    LayoutInflater inflater = LayoutInflater.from(parent.getContext());
    View rowRoot = inflater.inflate(R.layout.item_cast, parent, false);
    return new CastHolder(rowRoot);
  }

  @Override public void onBindViewHolder(CastHolder holder, int position) {
    MovieDetail.Cast cast = casts.get(position);
    if (cast.getUrlSmallImage() != null && !Utils.isEmpty(cast.getUrlSmallImage())) {
      GlideUtil.loadImg(context, cast.getUrlSmallImage(), holder.castImageView);
    }
    if (cast.getName() != null && !Utils.isEmpty(cast.getName())) {
      holder.castTextView.setText(cast.getName());
    }
  }

  @Override public int getItemCount() {
    return casts.size();
  }

  public class CastHolder extends RecyclerView.ViewHolder {

    @BindView(R.id.item_cast_iv_pic) CircleImageView castImageView;
    @BindView(R.id.item_cast_tv_name) TextView castTextView;

    CastHolder(View itemView) {
      super(itemView);
      ButterKnife.bind(this, itemView);
    }
  }
}

File: app-kotlin/src/main/java/net/mEmoZz/yts/kotlin/ui/detail/adapters/recycler/CastAdapter.kt
package net.mEmoZz.yts.kotlin.ui.detail.adapters.recycler

import android.content.Context
import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import kotlinx.android.synthetic.main.item_cast.view.item_cast_iv_pic
import kotlinx.android.synthetic.main.item_cast.view.item_cast_tv_name
import net.mEmoZz.yts.kotlin.R
import net.mEmoZz.yts.kotlin.data.models.MovieDetail
import net.mEmoZz.yts.kotlin.utilities.GlideUtil
import net.mEmoZz.yts.kotlin.utilities.Utils

/**
 * Authored by Mohamed Fathy on 08 Mar, 2018.
 * Contact: muhamed.gendy@gmail.com
 */

class CastAdapter(
    private val context: Context, private val casts: List<MovieDetail.Cast>
) : RecyclerView.Adapter<CastAdapter.CastHolder>() {

  override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): CastHolder {
    val inflater = LayoutInflater.from(parent.context)
    val rowRoot = inflater.inflate(R.layout.item_cast, parent, false)
    return CastHolder(rowRoot)
  }

  override fun onBindViewHolder(holder: CastHolder, position: Int) {
    val cast = casts[position]
    if (cast.urlSmallImage != null && !Utils.isEmpty(cast.urlSmallImage)) {
      GlideUtil.loadImg(context, cast.urlSmallImage, holder.itemView.item_cast_iv_pic)
    }
    if (cast.name != null && !Utils.isEmpty(cast.name)) {
      holder.itemView.item_cast_tv_name.text = cast.name
    }
  }

  override fun getItemCount(): Int {
    return casts.size
  }

  inner class CastHolder internal constructor(itemView: View) : RecyclerView.ViewHolder(itemView)
}
